src/codec/framed_read.rs

```diff
@@ -333,6 +90,256 @@ impl<T> FramedRead<T> {
     }
 }
 
+/// Decodes a frame.
+///
+/// This method is intentionally de-generified and outlined because it is very large.
+#[inline(never)]
```

Comment:
Is there a need to have this? Could we just leave that up to LLVM?

Reply:
I didn't actually test without this, will do and report back.